cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories_auth.go

```diff
@@ -192,19 +194,128 @@ func newAppRepositoryAuth(secret *k8scorev1.Secret,
-		if secret, err := typedClient.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{}); err != nil {
-			return nil, statuserror.FromK8sError("create", "secret", secret.GetName(), err)
+		secretName := secret.GetName()
+		newSecret, err = typedClient.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
+		if err != nil {
+			return nil, nil, statuserror.FromK8sError("create", "secret", secretName, err)
 		}
 	}
-	return secret, nil
+	if imagesPullSecret != nil {
+		secretName := imagesPullSecret.GetName()
+		newImgPullSecret, err = typedClient.CoreV1().Secrets(namespace).Create(ctx, imagesPullSecret, metav1.CreateOptions{})
+		if err != nil {
+			return nil, nil, statuserror.FromK8sError("create", "secret", secretName, err)
+		}
+	}
+	return newSecret, newImgPullSecret, nil
+}
+
+func validateDockerImagePullSecret(ctx context.Context,
+	typedClient kubernetes.Interface,
+	repoName types.NamespacedName,
+	secretName string) (*k8scorev1.Secret, error) {
+
+	if secret, err := typedClient.CoreV1().Secrets(repoName.Namespace).Get(ctx, secretName, metav1.GetOptions{}); err != nil {
+		return nil, statuserror.FromK8sError("get", "secret", secretName, err)
+	} else if secret.Type != k8scorev1.SecretTypeDockerConfigJson {
+		return nil, status.Errorf(codes.InvalidArgument, "Images Docker pull secret %s does not have valid type", secretName)
+	} else if _, ok := secret.Data[k8scorev1.DockerConfigJsonKey]; !ok {
+		return nil, status.Errorf(codes.InvalidArgument, "Images Docker pull secret %s does not have valid data", secretName)
+	} else {
+		return secret, nil
+	}
+}
+
+func imagesPullSecretName(repoName string) string {
+	return fmt.Sprintf("pullsecret-%s", repoName)
+}
+
+func newDockerImagePullSecret(ownerRepo types.NamespacedName, credentials *corev1.DockerCredentials) (secret *k8scorev1.Secret, isSameSecret bool, err error) {
+
+	secret = &k8scorev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: imagesPullSecretName(ownerRepo.Name),
+		},
+		Type: k8scorev1.SecretTypeDockerConfigJson,
+		Data: map[string][]byte{},
+	}
+
+	if credentials.Password == RedactedString {
+		isSameSecret = true
+	} else {
+		dockerConfig := &credentialprovider.DockerConfigJSON{
+			Auths: map[string]credentialprovider.DockerConfigEntry{
+				credentials.Server: {
+					Username: credentials.Username,
+					Password: credentials.Password,
+					Email:    credentials.Email,
+				},
+			},
+		}
+		dockerConfigJson, err := json.Marshal(dockerConfig)
+		if err != nil {
+			return nil, false, status.Errorf(codes.InvalidArgument, "Docker credentials are malformed")
+		}
+		secret.Data[k8scorev1.DockerConfigJsonKey] = dockerConfigJson
+	}
+
+	return secret, isSameSecret, nil
+}
+
+func deleteSecret(ctx context.Context, secretsInterface v1.SecretInterface, secretName string) error {
+	// Ignore action if secret didn't exist
+	if secret, _ := secretsInterface.Get(ctx, secretName, metav1.GetOptions{}); secret != nil {
```

Comment:
When performing a `.Get` operation, if not found, the struct being returned is an empty one, like &corev1.Secret{}, but it is not `nil`. 

Instead, I'd do:
```suggestion
	if _, err := secretsInterface.Get(ctx, secretName, metav1.GetOptions{}); err == nil {
```